src/core/glyf.js

```diff
@@ -0,0 +1,714 @@
+/* Copyright 2021 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const ON_CURVE_POINT = 1 << 0;
+const X_SHORT_VECTOR = 1 << 1;
+const Y_SHORT_VECTOR = 1 << 2;
+const REPEAT_FLAG = 1 << 3;
+const X_IS_SAME_OR_POSITIVE_X_SHORT_VECTOR = 1 << 4;
+const Y_IS_SAME_OR_POSITIVE_Y_SHORT_VECTOR = 1 << 5;
+const OVERLAP_SIMPLE = 1 << 6;
+
+const ARG_1_AND_2_ARE_WORDS = 1 << 0;
+const ARGS_ARE_XY_VALUES = 1 << 1;
+// const ROUND_XY_TO_GRID = 1 << 2;
+const WE_HAVE_A_SCALE = 1 << 3;
+const MORE_COMPONENTS = 1 << 5;
+const WE_HAVE_AN_X_AND_Y_SCALE = 1 << 6;
+const WE_HAVE_A_TWO_BY_TWO = 1 << 7;
+const WE_HAVE_INSTRUCTIONS = 1 << 8;
+// const USE_MY_METRICS = 1 << 9;
+// const OVERLAP_COMPOUND = 1 << 10;
+// const SCALED_COMPONENT_OFFSET = 1 << 11;
+// const UNSCALED_COMPONENT_OFFSET = 1 << 12;
+
+/*
+  GlypfTable object represents a glyf table containing glyph information:
+  - glyph header (xMin, yMin, xMax, yMax);
+  - contours if any;
+  - component if the glyph is a composite.
+   
+  It's possible to re-scale each glyph in order to have a new font which exactly
+  fit an other one: the goal is to be able to build some substitution font for
+  well-known fonts (Myriad, Arial, ...).
+
+  A full description of glyf table can be found here
+  https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6glyf.html
+*/
+class GlyfTable {
+  constructor(glyphs) {
+    this.glyphs = glyphs;
+  }
+
+  static parse({ glyfTable, isGlyphLocationsLong, locaTable, numGlyphs }) {
```

Comment:
Can't you simply let the `GlyfTable` constructor do this, and remove the static method, instead?